src/main/java/com/royalrangers/service/achievement/UserTestService.java

```diff
@@ -64,8 +69,27 @@ public void editUserTest(UserAchievementRequestDto params, Long id) {
         savedUserAchievement.setUpdateDate(new Date());
         String achievementState = params.getState();
         savedUserAchievement.setAchievementState(AchievementState.valueOf(achievementState));
-        Integer testId = params.getId();
-        savedUserAchievement.setTest(testService.getTestById(testId.longValue()));
         userTestRepository.saveAndFlush(savedUserAchievement);
+        Stream.of(UserAgeGroup.values()).forEach(ageGroups -> {
+            updateQuarterAchievements(UserAgeGroup.valueOf(ageGroups.toString()));
+        });
     }
+
+    public void updateQuarterAchievements(UserAgeGroup userAgeGroup) {
+        List<UserTest> userTestListForBeginner = userTestRepository.findByAchievementStateAndTest_UserAgeGroupsContains(AchievementState.APPROVED, new ArrayList<>(Arrays.asList(userAgeGroup)));
+        if (userTestListForBeginner.size() == 1 || userTestListForBeginner.size() == 5 || userTestListForBeginner.size() == 9 || userTestListForBeginner.size() == 13 ||
+                userTestListForBeginner.size() == 17 || userTestListForBeginner.size() == 21 || userTestListForBeginner.size() == 25 || userTestListForBeginner.size() == 29 ||
+                userTestListForBeginner.size() == 33 || userTestListForBeginner.size() == 37 || userTestListForBeginner.size() == 41 || userTestListForBeginner.size() == 45) {
+            userQuarterAchievementService.addUserQuarterAchievement(userAgeGroup);
```

Comment:
Move this if statement into the separate method. Give appropriate name for this method and the comment that will describe what is it.